azopsreference/3fc1081d-6105-4e19-b60c-1ec1252cf560 (3fc1081d-6105-4e19-b60c-1ec1252cf560)/contoso (contoso)/.AzState/Microsoft.Authorization_policyDefinitions-Deploy-Sql-IaaSExtension-For-AHUB.parameters.json

```diff
@@ -0,0 +1,124 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "input": {
+      "value": {
+        "Name": "Deploy-Sql-IaaSExtension-For-AHUB",
+        "ResourceType": "Microsoft.Authorization/policyDefinitions",
+        "SubscriptionId": null,
+        "Properties": {
+          "Description": "Deploy SQL IaaS Extension to enable SQL Server VMs with the Azure Hybrid Benefit",
+          "DisplayName": "Deploy-Sql-IaaSExtension-For-AHUB",
+          "Mode": "All",
+          "Parameters": { 
+            "tagName": {
+              "type": "String",
+              "metadata": {
+                "displayName": "SQL VM Tag Name",
+                "description": "Name of the tag that you use to identify SQL IaaS VMs to which Azure Hybrid Benefit will be applied, i.e., isSqlVM"
+              }
+            },
+            "sqlManagement": {
+              "type": "String",
+              "allowedValues": [
+                "Lightweight",
+                "Full"
+              ],        
+              "metadata": {
+                "displayName": "SQL Management",
+                "description": "Lightweight, or Full SQL Management. Note: Choosing Full SQL Management would restart the SQL Server in your VM"
+              }
+            }            
+          },
+          "PolicyRule": {
+            "if": {
+              "allOf": [
+                {
+                  "field": "type",
+                  "equals": "Microsoft.Compute/virtualMachines"
+                },
+                {
+                  "field": "[concat('tags[', parameters('tagName'), ']')]",
+                  "equals": "true"
+                }                
+              ]
+            },
+            "then": {
+              "effect": "deployIfNotExists",
+              "details": {
+                "type": "Microsoft.SqlVirtualMachine/SqlVirtualMachines",
+                "name": "[field('name')]",
```

Comment:
We can remove this line entirely as we don't want to check for a specific resource name of what needs to be installed.

Reply:
This line is actually required because it deploys a SqlIaasExtension for each VM which is qualified to have it. The SqlIaasExtension also takes the same name as the VM resource name.